src/App.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="main-content">
+    <the-memo-list
```

Comment:
再利用を前提としたコンポーネント名に `The` は不要かなと思いました 😅 `MemoList` で良いかと 👍

Reply:
言われてみればそうですね😱
1回しか使ってないから`The`を付けてみたのですが、もし今後アプリの内容を充実させる時にまた使うかもしれないからですね😄
Formの方は複数使う想定はないので`The`を付けてても問題ないということですね。

Reply:
すみません、 `Form` の方も `The` を外してもらう想定でした 😅 
コンポーネントである以上、再利用が前提なので、基本的に `The` を付けることはないですね 🙏

Reply:
分からなくなってきました😱
質問させてください🙇‍♂️

質問①
一応[スタイルガイド](https://v3.ja.vuejs.org/style-guide/#%E5%8D%98%E4%B8%80%E3%82%A4%E3%83%B3%E3%82%B9%E3%82%BF%E3%83%B3%E3%82%B9%E3%81%AE%E3%82%B3%E3%83%B3%E3%83%9B%E3%82%9A%E3%83%BC%E3%83%8D%E3%83%B3%E3%83%88%E5%90%8D-%E5%BC%B7%E3%81%8F%E6%8E%A8%E5%A5%A8)見て`The`を付けていたのですが、`Form`もこのスタイルには当てはまらないということでしょうか？それともこのスタイルガイドをあまり意識しないほうが良いということでしょうか？

以下引用です。
>常に 1 つのアクティブなインスタンスしか持たないコンポーネントは、1 つしか存在しえないことを示すために The というプレフィックスで始めるべきです。
これはそのコンポーネントが 1 つのページでしか使われないということを意味するのではなく、ページごとに 1 回しか使われないという意味です。これらのコンポーネントは、アプリケーション内のコンテキストではなく、アプリケーションに対して固有のため、決してプロパティを受け入れることはありません。もしプロパティを追加する必要があることに気づいたのなら、それは 現時点で ページごとに 1 回しか使われていないだけで、実際には再利用可能なコンポーネントだということを示すよい目印です。

最初この日本語文しか読んでなかったのですがプロパティって何？と思ったので[英文](https://v2.vuejs.org/v2/style-guide/?redirect=true#Single-instance-component-names-strongly-recommended)みたら`props`のことでした。

>These components never accept any props, since they are specific to your app, not their context within your app. If you find the need to add props, it’s a good indication that this is actually a reusable component that is only used once per page for now.

（これ読むと`The`を付けたコンポーネントは`props`を使ってはいけないとのことなので、`The`を付けて`props memos`まで使っていた私はそもそもスタイルガイドを理解していなかったことに・・・😞）

質問②
引用のこちらの部分、
「もしプロパティを追加する必要があることに気づいたのなら、それは 現時点で ページごとに 1 回しか使われていないだけで、実際には再利用可能なコンポーネントだということを示すよい目印です。」
なぜ`props`を使っていたら再利用可能なコンポーネントだと言えるのでしょうか？👀

長くなってしまいました🙇‍♂️
全く急ぎではないのでお時間ある時に教えてください。

Reply:
@kazumi3858 
混乱させてしまったようで申し訳ありませんでした 🙇 

①についてスタイルガイドに記載があるのですね 👀 これは私も知らなかったので、逆に勉強させられました 🙏 ありがとうございました。

②について、自分の @kazumi3858 さんと概ね同感です 😅 ページごとに1回だけしか使わないかどうか、と `props` があるかどうかは直接関係ないと思います。

----

おそらくここで筆者が言いたかったのは「 そのページに固有なコンポーネント(ヘッダやサイドバー等、仕様がどう転んでも1回しか登場し得ないコンポーネント)には `The` を付けてもいいよ。ただ、コンポーネントが `props` を受け取るということは、将来的に別の値を `props` に渡して再利用する可能性があるってことだから `The` は付けないでね」ということなのかなーと推察しました。

この考え方に照らし合わせると、 `MemoList` も `Form` も将来的に別の `props` を渡して再利用しうるので `The` は付けない方が良いと思いました 👍

Reply:
レビューありがとうございます🙏✨

スタイルガイドに関してはフィヨルドのプラクティスページに「目を通すように」と書いてあったので読みましたが、書いてなかったら私も存在さえ知らなかったと思います😅

>この考え方に照らし合わせると、 MemoList も Form も将来的に別の props を渡して再利用しうるので The は付けない方が良いと思いました 👍

たしかにMemo用途以外のフォームとして使うことがあるかも・・・と考えると納得です！😄
修正しました！
[複数単語のコンポーネント名](https://v3.ja.vuejs.org/style-guide/#%E8%A4%87%E6%95%B0%E5%8D%98%E8%AA%9E%E3%81%AE%E3%82%B3%E3%83%B3%E3%83%9B%E3%82%9A%E3%83%BC%E3%83%8D%E3%83%B3%E3%83%88%E5%90%8D-%E5%BF%85%E9%A0%88)に沿って`MemoForm`に変えました😸